src/csm_testing/tests/sat_functional/bootprep/test_bootprep.py

```diff
@@ -0,0 +1,416 @@
+                          f"Standard Error: \n{err.stderr.decode()}\n")
+            raise
+
+        return result
+
+
+class TestBootprepCreateConfigs(BootprepRunTestCase):
+    """Tests for creating CFS configurations using `sat bootprep run`"""
+
+    def test_no_configs(self):
+        """Test that a file with an empty list of configs creates no configs"""
+        result = self.run_bootprep('no-configs.yaml', '--format json')
+
+        report = json.loads(result.stdout.decode())
+        self.assertEqual({}, report)
+        self.assert_in_log_messages(
+            "INFO",
+            "Nothing to create in collection of CFS configurations",
+            result.stderr.decode()
+        )
+
+    def test_no_layers(self):
+        """Test creating a CFS configuration with no layers"""
+        result = self.run_bootprep('no-layers-config.yaml', '--format json')
+
+        report = json.loads(result.stdout.decode())
+        self.assertEqual(1, len(report['configurations']))
+        self.assertEqual(f'{self.test_prefix}-no-layers', report['configurations'][0]['name'])
+
+        self.items_to_delete['configurations'].append(f'{self.test_prefix}-no-layers')
+
+    @skip_test_if_csm_var_missing(['version', 'commit_hash'])
+    def test_product_layers(self):
+        """Test creating multiple CFS configurations using product-based layers"""
+        result = self.run_bootprep('product-layers-config.yaml', '--format json')
+
+        report = json.loads(result.stdout.decode())
+        self.assertEqual(5, len(report['configurations']))
+
+        # Query CFS for the configurations and ensure they exist
+        # Validate certain elements of the created CFS configurations including:
+        #   - configuration names
+        #   - clone URLs
+        #   - commit hashes
+
+        for config in report['configurations']:
+            self.items_to_delete['configurations'].append(config['name'])
+
+    def test_git_layers(self):
+        """Test creating multiple CFS configurations with git-based layers"""
+        result = self.run_bootprep('git-layers-config.yaml', '--format json')
+
+        report = json.loads(result.stdout.decode())
+        self.assertEqual(3, len(report['configurations']))
+
+        # Query CFS for the configurations and ensure they exist
+        # Validate certain elements of the created CFS configurations including:
+        #   - configuration names
+        #   - clone URLs
+        #   - commit hashes
+
+        for config in report['configurations']:
+            self.items_to_delete['configurations'].append(config['name'])
+
+    def test_special_parameters(self):
+        """Test creating a CFS configuration with special parameters"""
+        result = self.run_bootprep('special-parameters-config.yaml', '--format json')
+
+        report = json.loads(result.stdout.decode())
+        self.assertEqual(1, len(report['configurations']))
+        self.assertEqual(f'{self.test_prefix}-special-parameters',
```

Comment:
The name should be same as you defined in special-parameters-config.yaml `- name: "{{test.prefix}}-csm-special-parameters-layers"`
```suggestion
        self.assertEqual(f'{self.test_prefix}-csm-special-parameters-layer',
```